src/ssl.c

```diff
@@ -0,0 +1,64 @@
+/*
+  Axel -- A lighter download accelerator for Linux and other Unices
+
+  Copyright 2001-2007 Wilmer van der Gaast
+  Copyright 2008      Y Giridhar Appaji Nag
+  Copyright 2008-2009 Philipp Hagemeister
+  Copyright 2015      Joao Eriberto Mota Filho
+  Copyright 2016      Ivan Gimenez
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+/* SSL interface */
+
+#include "axel.h"
+
+static SSL_CTX *ssl_ctx = NULL;
+
+void ssl_init( void )
+{
+	if( ssl_ctx != NULL )
+		return;
+
+	SSL_library_init();
+	SSL_load_error_strings();
+
+	ssl_ctx = SSL_CTX_new( SSLv23_client_method() );
```

Comment:
Actually, `TLS_client_method()` isn't available yet in any released openssl versions. It's in master, but not 1.0.2.

https://www.openssl.org/docs/man1.0.2/ssl/SSL_CTX_new.html

It would seem `SSLv23_client_method()` is still the preferred option until then.